### PR TITLE
display ip from 'X-Real-IP' header if behind proxy

### PR DESCRIPTION
**Pull request by bbx-github**
Before localhost ip was displayed. I hope nothing else breaks.

### CODE REVIEW

**Review by TheJJ — CHANGES_REQUESTED** [review]
good idea. I'd suggest to implement it differently though:
add a new function to the sample config file, called `def get_ip(source_ip, headers)`, where the mapping can be done in any fancy way. As default, sample implementation would be:

``` python
def get_ip(source_ip, headers):
    if 'X-Real-IP' in headers:
        source_ip = headers['X-Real-IP']
    return source_ip
```

This function is then called in line 85 of the `handle_request` method. That way, IP mappings can be customized even by a database easily, or the header can be renamed, etc.

**Comment by bbx-github** [comment]
Does this match your expectations? It still has this `if headers` to not call `get_ip` and get an exception.

File: sftdyn/server.py [hunk 1]
```diff
@@ -83,6 +83,11 @@ def __init__(self, addr, clients, associations, nsupdatecommands, tls=None):
             (status, httpcode) after examining the key
         """
 
+        if headers:
+            ip = self.get_ip(ip, headers)
+        else:
+            headers = dict()
```
Comment (TheJJ) [inline sftdyn/server.py:88]: as the `get_ip` function also works if the headers are empty, you can just do a call like this:
``` python
if headers is None:
    headers = dict()
ip = self.get_ip(ip, headers)
```

File: etc/sample.conf [hunk 2]
```diff
@@ -59,5 +56,17 @@ def nsupdatecommands(host, new_ip, headers):
 
     return cmds
 
+def get_ip(source_ip, headers):
```
Comment (TheJJ) [inline etc/sample.conf]: as this happens before the `nsupdatecommands`, I'd move this function above it.

**Review by TheJJ — CHANGES_REQUESTED** [review]

**Comment by bbx-github** [comment]
Was this really necessary? I thought the config gets imported by `args.py` with the result that function order does not matter.

**Comment by TheJJ** [comment]
That's true, but as this is an example config file it's easier to understand what happens if the functions are written in their logical order.

**Review by TheJJ — APPROVED** [review]
looking good now :)

**Review by TheJJ — CHANGES_REQUESTED** [review]
ah, you also need to register the function in `Server`, just like it is done with `nsupdatecommands`.

**Comment by bbx-github** [comment]
Oh - yes. Obviously I did not understand how it works. And obviously I did not use this code but my simple fix. Anyway I like to add this feature. So I will do a installation of this code for testing.
I think this should be an optional configuration because with a simple setup it is not needed.

**Comment by bbx-github** [comment]
I did setup setup a installation for testing and now it works with and without `def get_ip` in config.

**Review by TheJJ — APPROVED** [review]
nice!

**Pull request merged**